src/Utils/InterfaceValidator.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Utils;
+
+use WordPress\AiClient\Providers\Models\Contracts\ModelInterface;
+use WordPress\AiClient\Providers\Models\ImageGeneration\Contracts\ImageGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationOperationModelInterface;
+use WordPress\AiClient\Providers\Models\TextGeneration\Contracts\TextGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionOperationModelInterface;
+
+/**
+ * Utility class for validating model interface implementations.
+ *
+ * Centralizes interface validation logic to reduce code duplication
+ * and provide consistent error messages across the AI Client.
+ *
+ * @since n.e.x.t
+ */
+class InterfaceValidator
+{
+    /**
+     * Validates that a model implements TextGenerationModelInterface.
+     *
+     * @since n.e.x.t
+     *
+     * @param ModelInterface $model The model to validate.
+     * @return void
```

Comment:
If you add the [phpstan-assert](https://phpstan.org/writing-php-code/narrowing-types#custom-type-checking-functions-and-methods) tag, then it will inform the type system that this method is validating the instance. This will make it so you don't have to ignore PhpStan errors downstream.
```suggestion
     * @phpstan-assert TextGenerationModelInterface $model	
     * @param ModelInterface $model The model to validate.
     * @return void
```

Apply something like this to the other methods as well. 😄

Reply:
@JasonTheAdams Great suggestion! I've added `@phpstan-assert` annotations to all the validation methods in `InterfaceValidator`resolved through https://github.com/WordPress/php-ai-client/pull/55/commits/c75f01acbdfd43d206d793abca64c0487b05495a